qmcpy/true_measure/matern.py

```diff
@@ -0,0 +1,140 @@
+                        [0.01  0.01  0.01  ... 0.002 0.002 0.002]
+                        ...
+                        [0.002 0.002 0.002 ... 0.01  0.01  0.01 ]
+                        [0.002 0.002 0.002 ... 0.01  0.01  0.01 ]
+                        [0.001 0.002 0.002 ... 0.01  0.01  0.01 ]]
+        decomp_type     PCA
+    >>> m3 = Matern(Lattice(dimension = 31,seed=7), x_values, length_scale = 0.5, nu = 3.5, variance = 0.01, mean=mean, decomp_type = 'Cholesky')
+    >>> m3
+    Matern (TrueMeasure Object)
+        mean            [1.1 1.1 1.1 ... 1.1 1.1 1.1]
+        covariance      [[0.01  0.01  0.01  ... 0.002 0.002 0.001]
+                        [0.01  0.01  0.01  ... 0.002 0.002 0.002]
+                        [0.01  0.01  0.01  ... 0.002 0.002 0.002]
+                        ...
+                        [0.002 0.002 0.002 ... 0.01  0.01  0.01 ]
+                        [0.002 0.002 0.002 ... 0.01  0.01  0.01 ]
+                        [0.001 0.002 0.002 ... 0.01  0.01  0.01 ]]
+        decomp_type     CHOLESKY
+
+     References:
+
+        [1] scikit-learn developers, Matern kernel. 
+        https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.kernels.Matern.html.
+        Accessed 2023.
+
+        [2] Abramowitz and Stegun (1972). Handbook of Mathematical Functions with Formulas, Graphs, 
+        and Mathematical Tables. ISBN 0-486-61272-4. (Accessed through Wikipedia, 
+        https://en.wikipedia.org/wiki/Mat%C3%A9rn_covariance_function.)
+    """
+
+    def __init__(self, sampler, x_values, length_scale = 1.0, nu = 1.5, variance = 1.0, mean = [], decomp_type='PCA'):
+        """
+        Matern kernel: calculates covariance over a metric space based only on the distance between points.
+        More information can be found at [1].
+        
+        Args:
+            sampler (DiscreteDistribution/TrueMeasure): A 
+                discrete distribution from which to transform samples or a
+                true measure by which to compose a transform. 
+            x_values (array): The positions of points on a metric space.
+            length_scale (float): Determines "peakiness", or how correlated 
+                two points are based on their distance.
+            nu (float): The "smoothness" of the Matern function. e.g. nu=1.5
+                implies a once-differentiable function, while nu=2.5 implies twice
+                differentiability. Meanwhile, when nu=0.5, the Matern kernel equals
+                the RBF kernel, while as nu approaches inf, it equals the absolute 
+                exponential kernel. Note that nu values not in [0.5, 1.5, 2.5, inf] 
+                will be ~10x slower to run.
+            variance (float): The variance (or the diagonal elements of the
+                covariance matrix) of the distribution at each point. Implemented
+                as a constant factor of the Matern covariance. The default 
+                value is 1.0.
```

Comment:
I would call this a scaling factor since it multiplies *all* values of the kernel matrix by a constant.